src/components/TimePicker/TimePicker.js

```diff
@@ -145,35 +159,37 @@ function TimePicker({forwardedRef, defaultValue, onSubmit, onInputChange}) {
                 newHour = `${formattedText[0]}${formattedText[1]}`;
                 newSelection = 1;
             }
-        } else if (selectionHour.start === 1) {
-            // Case when the cursor is at the second position.
+        } else if (selectionHour.start === 1 && selectionHour.end === 1) {
+            // The cursor is at the second position.
             const formattedText = `${filteredText[0]}${filteredText[1]}`;
 
             if (filteredText.length < 2) {
                 // If we remove a value, prepend 0.
-                newHour = `0${text}`;
+                newHour = `0${filteredText}`;
                 newSelection = 0;
                 // If the second digit is > 2, replace the hour with 0 and the second digit.
             } else if (formattedText > 12 && formattedText <= 24) {
                 newHour = String(formattedText - 12).padStart(2, '0');
                 newSelection = 2;
                 setAmPmValue(CONST.TIME_PERIOD.PM);
             } else if (formattedText > 24) {
-                newHour = `0${text[1]}`;
+                newHour = `0${filteredText[1]}`;
                 newSelection = 2;
             } else {
-                newHour = `${text[0]}${text[1]}`;
+                newHour = `${filteredText[0]}${filteredText[1]}`;
                 setHours(newHour);
                 newSelection = 2;
             }
-        } else if (selectionHour.start === 2 && selectionHour.end === 2) {
-            // Case when the cursor is at the end and no text is selected.
-            if (filteredText.length < 2) {
-                newHour = `${text}0`;
-                newSelection = 1;
-            } else {
-                newSelection = 2;
-            }
+        } else if (filteredText.length <= 1 && filteredText < 2) {
+            newHour = `${filteredText}0`;
+            newSelection = 1;
+        } else if (filteredText > 12 && filteredText <= 24) {
+            newHour = String(filteredText - 12).padStart(2, '0');
+            newSelection = 2;
+            setAmPmValue(CONST.TIME_PERIOD.PM);
+        } else if (filteredText.length <= 2) {
+            newHour = String(Math.min(filteredText, 12)).padStart(2, '0');
+            newSelection = 2;
         }
```

Comment:
Can you add a comment for each case here?

Reply:
Sure.
We also have a discrepancy in naming: `hours` and `minute`. Would you mind if I made the naming consistent and grammatically correct by changing to `hours` and `minutes`?

Reply:
That sounds good to me.